### PR TITLE
cut/paste _Timestamp+Timestamp to tslibs.timestamps

### PR DESCRIPTION
**Pull request by jbrockmendel**
Then remove unused imports in tslibs.

If/when this is merged, it'll be official: we've taken tslib from a 5500 line beast down to ~1k.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -601,6 +602,14 @@ def pxd(name):
                      '_libs/tslibs/nattype'],
         'depends': np_datetime_headers,
         'sources': np_datetime_sources},
+    '_libs.tslibs.timestamps': {
```
Comment (jreback) [inline setup.py]: tslib cimports timestamps

File: pandas/_libs/tslibs/timestamps.pyx [hunk 2]
```diff
@@ -0,0 +1,981 @@
...
+        return (self.time() != _zero_time
+                or self.tzinfo is not None
+                or self.nanosecond != 0)
+
+    def to_julian_date(self):
+        """
+        Convert TimeStamp to a Julian Date.
+        0 Julian date is noon January 1, 4713 BC.
+        """
+        year = self.year
```
Comment (jreback) [inline pandas/_libs/tslibs/timestamps.pyx]: can cdef these (TODO)

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
> If/when this is merged, it'll be official: we've taken tslib from a 5500 line beast down to ~1k.

sure but still some things things to move to the subs :> (I know already on your TODO).

in the end of the day tslibx.pyx could/should just be imports :>

File: setup.py (hunk 1 above)
Comment (jbrockmendel) [reply setup.py]: Fixing now.

File: pandas/_libs/tslibs/timestamps.pyx (hunk 2 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/timestamps.pyx]: Yah, in fact a bunch of stuff can be moved up to `_Timestamp`.

**Comment by jreback** [comment]
lgtm. ping on green.

**Comment by jbrockmendel** [comment]
Ping

**Comment by jreback** [comment]
rebase

**Comment by jbrockmendel** [comment]
ping

File: pandas/_libs/tslib.pyx [hunk 3]
```diff
@@ -847,331 +200,6 @@ cdef inline bint _check_all_nulls(object val):
...
-    def asm8(self):
-        return np.datetime64(self.value, 'ns')
-
-    def timestamp(self):
-        """Return POSIX timestamp as float."""
-        # py27 compat, see GH#17329
-        return round(self.value / 1e9, 6)
-
-
 cdef PyTypeObject* ts_type = <PyTypeObject*> Timestamp
```
Comment (jreback) [inline pandas/_libs/tslib.pyx]: is this needed? remove later?

Comment (jbrockmendel) [reply pandas/_libs/tslib.pyx]: Still used in a couple places, we might be able to remove it later.  Let’s get this and libmissing done and then take a look.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**